dhall/src/Dhall/Main.hs

```diff
@@ -542,9 +545,14 @@ noHeaders i =
 -- | Run the command specified by the `Options` type
 command :: Options -> IO ()
 command (Options {..}) = do
-    let characterSet = case ascii of
-            True  -> ASCII
-            False -> Unicode
+    let chosenCharacterSet =
+            case (ascii, unicode) of
+                -- If both flags are set, default to ASCII
+                (True, _) -> Just ASCII
+                (_, True) -> Just Unicode
+                _         -> Nothing
```

Comment:
It's actually possible to ensure that the flags are mutually exclusive.  The trick is to replace this:

```haskell
    <*> switch "ascii" "Format code using only ASCII syntax"
    <*> switch "unicode" "Format code using only Unicode syntax" 
````

… with this:

```haskell
  <*> parseCharacterSet
  …
  where
    parseCharacterSet =
            flag' (Just Unicode) (long "unicode" <> help "Format code using only Unicode syntax")
        <|> flag' (Just ASCII) (long "ascii" <> help "Format code using only ASCII syntax")
        <|> pure Nothing
```

Reply:
That's a good idea! :smile: 

Implementing this change made me realise that we are still using the old `--ascii` switch for other things than just `dhall format`. We seems to be using it for pretty much every command. Should we also infer from the input in these other cases? Or should we limit this to `dhall format`? I should have thought about this earlier... :sweat_smile:

Reply:
The behavior should probably be the same for all of the commands, not just `dhall format`

Reply:
I believe I've now made it infer the CharacterSet in all commands where we could already specify which one we wanted. I think there are some remaining commands where it wasn't configurable before like `dhall diff` (where it always uses Unicode). Should I look into making them configurable as well?

Reply:
@basile-henry: You can if you would like to